crossing-repair.lic

```diff
@@ -33,6 +99,69 @@ class CrossingRepair
     fput('sort auto head') if settings.sort_auto_head
   end
 
+  # This method is designed to prep for repair by checking if we have a bank specified for the
+  # town in base-town.yaml, and checking if we have leather and/or metal items to repair and if
+  # there is a requisite metal or leather repair person specified in base-town.yaml.
+  # If we don't have the proper specifications in base-town.yaml, warn user, direct to submit issue.
+  # Returns: the hash of items to repair at either metal/leather repairer.
+  def repair_prep(town, settings)
+
+    DRC.message("Validating bank and repairers...") if $debug_mode_crossing_repair
+
+    # Populate a hash of leather items to repair, and metal items to repair, based on user's
+    # yaml list of equipment. While having to repair different items at leather vs. metal
+    # repair is deprecated, we still need to support the legacy settings.
+    leather_items_to_repair = @equipment_manager.items.select(&:leather).reject(&:skip_repair)
+    metal_items_to_repair = @equipment_manager.items.reject(&:leather).reject(&:skip_repair)
+
+    # Updates to DR now allow all items to be repaired at any repairer - whether historically
+    # a metal or leather repairer. In order to maintain legacy code functionality, we'll still
+    # accept specification that an item is leather or metal, but we'll be smart about towns
+    # that have only one type of repairer, and repair all items when we can.
+    leather_repairer = @hometown_data['leather_repair'] || @hometown_data['metal_repair']
+    metal_repairer = @hometown_data['metal_repair'] || @hometown_data['leather_repair']
+
+    # If we have @skip_bank specified in our yaml, we may not need a bank as long as we have
+    # enough money on hand to meet or exceed our yaml repair_withdrawal_amount. Otherwise,
+    # check for a bank in the specified town. Withdraw money if needed.
+    unless @skip_bank && (DRCM.wealth(settings['hometown']) >= @repair_withdrawal_amount)
```

Comment:
I'll need to rewrite this to handle situations where we don't know the currency of the specified town. My idea here is, if we have enough money on hand of the right currency to meet or exceed our yaml `repair_withdrawal_amount`, _and_ our yaml tells us to skip depositing excess coins at the end, then who cares whether we know if there is a bank to access for repairs? Just go repair.

Reply:
I opted to require knowing the currency of the specified repair town. Thoughts?